### PR TITLE
Add README.md Trademarks

### PR DESCRIPTION
**Pull request by annafil**
# Description

Adds trademark icons to respective Apache project names in first use, and trademark attribution statement to the bottom of the document.

Also updates README links to use the polaris.io domain name, while I'm in here 😉 

Partially Addresses #48 

## Type of change

Please delete options that are not relevant.

- [x] This change requires a documentation update

# How Has This Been Tested?

README.md change can be previewed in this PR

# Checklist:

Please delete options that are not relevant.

- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,12 +31,12 @@ Polaris Catalog is open source under an Apache 2.0 license.
...
 
-API docs are hosted via Github Pages at https://polaris-catalog.github.io/polaris. All updates to the main branch
+API docs are hosted via Github Pages at https://polaris.io. All updates to the main branch
 update the hosted docs.
 
-The Polaris management API docs are found [here](https://polaris-catalog.github.io/polaris/index.html#tag/polaris-management-service_other)
+The Polaris management API docs are found [here](https://polaris.io/index.html#tag/polaris-management-service_other)
 
-The open source Iceberg REST API docs are found [here](https://polaris-catalog.github.io/polaris/index.html#tag/Configuration-API)
+The open source Iceberg REST API docs are found [here](https://polaris.io/index.html#tag/Configuration-API)
```
Comment (flyrain) [inline README.md:39]: Minor: not necessarily related to this PR, but we may call it "Apache Iceberg" or "Iceberg" instead of "open source Iceberg"

**Review by flyrain — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 # Polaris Catalog
 
-<a href="https://www.snowflake.com/blog/polaris-catalog-open-source/" target="_blank">Polaris Catalog</a> is an open source catalog for Apache Iceberg. Polaris Catalog implements Iceberg’s open <a href="https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml" target="_blank">REST API</a> for multi-engine interoperability with Apache Doris, Apache Flink, Apache Spark, PyIceberg, StarRocks and Trino. 
+<a href="https://www.snowflake.com/blog/polaris-catalog-open-source/" target="_blank">Polaris Catalog</a> is an open source catalog for Apache Iceberg :tm:. Polaris Catalog implements Iceberg’s open <a href="https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml" target="_blank">REST API</a> for multi-engine interoperability with Apache Doris :tm:, Apache Flink :tm:, Apache Spark :tm:, PyIceberg, StarRocks and Trino.
```
Comment (snazy) [inline README.md:21]: It's actually Apache Flink®

File: README.md (hunk 1 above)
Comment (annafil) [reply README.md:39]: Good catch! Updating :)

File: README.md [hunk 3]
```diff
@@ -179,3 +179,7 @@ You must run the following as the first query in your spark-sql shell to actuall
 ```
 use polaris;
 ```
+
+### Trademark Attribution 
+
+_Apache Iceberg, Apache Spark, Apache Flink and Apache Doris are trademarks of The Apache Software Foundation._
```
Comment (jbonofre) [inline README.md:185]: We should use the full trademark attribution:

```
Apache Iceberg, Iceberg, Apache Spark, Spark, Apache Flink, Flink, Apache Doris, Doris, Apache, the Apache feather logo, the Apache Iceberg project logo, the Apache Spark project logo, the Apache Flink project logo, and the Apache Doris project logo are either registered trademarks or trademarks of The Apache Software Foundation. Copyright © 2024 The Apache Software Foundation
```

**Review by jbonofre — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (annafil) [reply README.md]: Fixed, thank you!

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (annafil) [reply README.md]: Done and done!

File: README.md [hunk 4]
```diff
@@ -18,7 +18,7 @@
 
 # Polaris Catalog
 
-<a href="https://www.snowflake.com/blog/polaris-catalog-open-source/" target="_blank">Polaris Catalog</a> is an open source catalog for Apache Iceberg. Polaris Catalog implements Iceberg’s open <a href="https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml" target="_blank">REST API</a> for multi-engine interoperability with Apache Doris, Apache Flink, Apache Spark, PyIceberg, StarRocks and Trino. 
+<a href="https://www.snowflake.com/blog/polaris-catalog-open-source/" target="_blank">Polaris Catalog</a> is an open source catalog for Apache Iceberg :tm:. Polaris Catalog implements Iceberg’s open <a href="https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml" target="_blank">REST API</a> for multi-engine interoperability with Apache Doris :tm:, Apache Flink® , Apache Spark :tm:, StarRocks and Trino.
```
Comment (jbonofre) [inline README.md:21]: Sorry, I wasn't clear.

We should use `registered` symbol here, not `trademark` (Apache Flink is correct, not the others).

See https://www.apache.org/foundation/marks/faq/

**Review by flyrain — APPROVED** [review]
LGTM. Thanks @annafil !

**Pull request merged**